File: samples/python-samples/python_samples/bedrock_data_automation_stack.py
from aws_cdk import (
    Stack,
    CfnOutput,
    aws_apigateway as apigw
)
from constructs import Construct
from aws_solutions_constructs.aws_apigateway_lambda import ApiGatewayToLambda
from cdklabs.generative_ai_cdk_constructs import BedrockDataAutomation

class BedrockDataAutomationStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # Create BedrockDataAutomation construct
        bda = BedrockDataAutomation(self, 'cb',
            is_custom_bda_blueprint_required=True,
            is_bda_project_required=True,
            is_bda_invocation_required=True,
            is_status_required=True
        )

        # Create CFN outputs for bucket names
        CfnOutput(self, 'inputbucketname', value=bda.input_bucket.bucket_name)
        CfnOutput(self, 'outputbucketname', value=bda.output_bucket.bucket_name)

        # Create API Gateway integrations for each Lambda function
        blueprint_api = ApiGatewayToLambda(self, 'CreateBlueprintApi',
            existing_lambda_obj=bda.blueprint_lambda_function,
            api_gateway_props=apigw.RestApiProps(
                rest_api_name='createBluePrintPython'
            )    
        )

        invoke_api = ApiGatewayToLambda(self, 'InvokeApi',
            existing_lambda_obj=bda.bda_invocation_lambda_function,
             api_gateway_props=apigw.RestApiProps(
                rest_api_name='createBluePrintPython'
            ) 
        )

        project_api = ApiGatewayToLambda(self, 'bdaProjectApi',
            existing_lambda_obj=bda.bda_project_lambda_function,
             api_gateway_props=apigw.RestApiProps(
                rest_api_name='createBDAprojectPython'
            ) 
        )

        result_status_api = ApiGatewayToLambda(self, 'bdaResultApi',
            existing_lambda_obj=bda.bda_result_statu_lambda_function,
             api_gateway_props=apigw.RestApiProps(
                rest_api_name='bdaResultStatusPython'
            ) 
        )

File: samples/bedrock-data-automation/lib/bedrock-data-automation-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as events from 'aws-cdk-lib/aws-events';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import * as sqs from 'aws-cdk-lib/aws-sqs';
import { EventbridgeToLambda } from '@aws-solutions-constructs/aws-eventbridge-lambda';
import * as genai from '@cdklabs/generative-ai-cdk-constructs';

export class BedrockDataAutomationStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    
    const bda = new genai.BedrockDataAutomation(this, 'cb', {
      isCustomBDABlueprintRequired: true,
      isBDAProjectRequired: true,
      isBDAInvocationRequired: true,
      isStatusRequired: true,
    });
 
    new cdk.CfnOutput(this, 'inputbucketname', { value: bda.inputBucket.bucketName });
    new cdk.CfnOutput(this, 'outputbucketname', { value: bda.outputBucket.bucketName });


    const bdaProjectFunctio = bda.bdaProjectLambdaFunction

    // create BDA blueprint
    const bluePrintFunction = bda.blueprintLambdaFunction
    const blueprintEventbridge = new EventbridgeToLambda(this, 'CreateBlueprintEventRule', {
      existingLambdaObj: bluePrintFunction,
      eventRuleProps: {
        eventPattern: {
          source: ['custom.bedrock.blueprint'],
          detailType: ['Bedrock Blueprint Request'],
        }
      },
    });

  

//  Invoke Data processing
const invokeFunction = bda.bdaInvocationLambdaFunction

const invokeEventbridge = new EventbridgeToLambda(this, 'InvokeEventRule', {
  existingLambdaObj: invokeFunction, // Your existing Lambda function
  eventRuleProps: {
    eventPattern: {
      source: ['custom.bedrock.blueprint'],
      detailType: ['Bedrock Invoke Request'],
      
    }
  },

});

const bdaProjectFunction = bda.bdaProjectLambdaFunction

const bdaProjectEvent =  new EventbridgeToLambda(this, 'bdaProjectRule', {
  existingLambdaObj: bdaProjectFunction, 
  eventRuleProps: {
    eventPattern: {
      source: ['custom.bedrock.blueprint'],
      detailType: ['Bedrock Project Request'],
      
    }
  },

  

}); 


const bdaResultStatusFunction = bda.bdaResultStatuLambdaFunction

const bdaResultEvent =  new EventbridgeToLambda(this, 'bdaResultRule', {
  existingLambdaObj: bdaResultStatusFunction, 
  eventRuleProps: {
    eventPattern: {
      source: ['custom.bedrock.blueprint'],
      detailType: ['Bedrock Result Status'],
      
    }
  },
  

}); 

  }
}


// Example event structure
// const event = {
//   detail: {
//     input_file_name: "your-input-file.txt"
//   },
//   detail-type: "FileProcessingEvent",
//   source: "custom.fileprocessing"
// }
